content/docs/2.7/scalers/gcp-stackdriver.md

```diff
@@ -0,0 +1,125 @@
++++
+title = "Google Cloud Platform Stackdriver"
+availability = "2.7+"
+maintainer = "Community"
+description = "Scale applications based on a metric obtained from Stackdriver."
+layout = "scaler"
+go_file = "gcp_stackdriver_scaler"
++++
+
+### Trigger Specification
+
+This specification describes the `gcp-stackdriver` trigger for GCP Stackdriver. It scales based on a metric obtained from issuing a query to Stackdriver.
+
+```yaml
+triggers:
+- type: gcp-stackdriver
+  metadata:
+    projectId: my-project-id
+    filter: 'metric.type="storage.googleapis.com/network/received_bytes_count" AND resource.type="gcs_bucket" AND metric.label.method="WriteObject" AND resource.label.bucket_name="my-gcp-bucket"'
+    metricName: my-gcp-bucket
```

Comment:
Can we please not force users to specify `metricName`, we should generate it instead.

Reply:
I know but there isn't any other parameter I can deduce a reasonable metric from. project_id is too generic and can collide with another such scaler